news/2016-10-10-tip2-devmode.asciidoc

```diff
@@ -0,0 +1,113 @@
+---
+layout: news
+title: Weld Tip &num;2 - Development Mode
+author: Martin Kouba
+priority: 1
+change_frequency: weekly
+excerpt: Meet the Weld Probe - a development tool which allows to inspect CDI components at runtime.
+desc: Weld Tip #2 - Development Mode
+tags: [tips]
+---
+:linkattrs:
+
+In this article we're going to cover the link:http://docs.jboss.org/weld/reference/latest/en-US/html/devmode.html[special mode for application development, window="_blank"].
+When enabled, a built-in tool called *Probe* provides a detailed view into internals of a CDI application.
+Probe makes it easy to inspect:
+
+* bean archives
+* beans and their properties such as qualifiers, stereotypes and name
+* a graph of bean dependencies
+* observers and producers declared by beans
+* interceptors and decorators bound to a bean
+* extensions
+* information about current contextual instances
+* tracking of invocations and invocation trees
+* tracking of fired events
+* Weld configuration
+* and more!
+
+Before you start further reading, please try out the link:http://probe-weld.itos.redhat.com/weld-numberguess/weld-probe[live demo hosted on OpenShift, window="_blank"] and feel the developer experience Weld is offering.
```

Comment:
I am gonna suggest a slightly different version here:
`Before you start reading, you might want to visit our live demo hosted on OpenShift and give it go yourself to truly see what Probe can offer.`